src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -143,10 +147,11 @@ public void AddOrUpdate(TKey key, TValue value)
         /// <summary>Removes a key and its value from the table.</summary>
         /// <param name="key">key to remove. May not be null.</param>
         /// <returns>true if the key is found and removed. Returns false if the key was not in the dictionary.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
         /// <remarks>
-        /// The key may get garbage collected during the Remove() operation. If so,
-        /// Remove() will not fail or throw, however, the return value can be either true or false
-        /// depending on who wins the race.
+        /// The key may get garbage collected during the <see cref="Remove(TKey)"/> operation. If so,
```

Comment:
Unless Remove is inlined, and presumably today it can't be due to the `lock` resulting in a try/finally, how could the key be garbage collected during the Remove call?

Reply:
Not sure, this comment was there before this change. Perhaps we should ask someone from the GC team?

Reply:
Updated in fd7e57ec9aa622bda99a1e3db3b8fa6e4f0a26c2! 🙂

Reply:
Is this something we will keep track of with any relevant future JIT/GC changes? If not, we should probably just add a `GC.KeepAlive` at the end of the method imo to ensure it works as it does today forevermore. These are basically free these days iirc anyway.

Reply:
Is it even needed though? Like, why would we want to keep the key alive in the first place? If it could get collected right after the call it means the value might as well be dropped earlier if possible, because no caller is going to observe it anyway. To me it just seems like this wouldn't be observable either way, so might as well not mention it.